packages/react-reconciler/src/ReactStrictModeWarnings.js

```diff
@@ -129,175 +42,264 @@ if (__DEV__) {
+    if (pendingUNSAFE_ComponentWillMountWarnings.length > 0) {
+      pendingUNSAFE_ComponentWillMountWarnings.forEach(fiber => {
+        UNSAFE_componentWillMountUniqueNames.add(
+          getComponentName(fiber.type) || 'Component',
+        );
+        didWarnAboutUnsafeLifecycles.add(fiber.type);
+      });
+      pendingUNSAFE_ComponentWillMountWarnings = [];
+    }
+
+    const componentWillReceivePropsUniqueNames = new Set();
+    if (pendingComponentWillReceivePropsWarnings.length > 0) {
+      pendingComponentWillReceivePropsWarnings.forEach(fiber => {
+        componentWillReceivePropsUniqueNames.add(
+          getComponentName(fiber.type) || 'Component',
+        );
+        didWarnAboutUnsafeLifecycles.add(fiber.type);
+      });
+
+      pendingComponentWillReceivePropsWarnings = [];
+    }
+
+    const UNSAFE_componentWillReceivePropsUniqueNames = new Set();
+    if (pendingUNSAFE_ComponentWillReceivePropsWarnings.length > 0) {
+      pendingUNSAFE_ComponentWillReceivePropsWarnings.forEach(fiber => {
+        UNSAFE_componentWillReceivePropsUniqueNames.add(
+          getComponentName(fiber.type) || 'Component',
+        );
+        didWarnAboutUnsafeLifecycles.add(fiber.type);
+      });
+
+      pendingUNSAFE_ComponentWillReceivePropsWarnings = [];
+    }
+
+    const componentWillUpdateUniqueNames = new Set();
+    if (pendingComponentWillUpdateWarnings.length > 0) {
+      pendingComponentWillUpdateWarnings.forEach(fiber => {
+        componentWillUpdateUniqueNames.add(
+          getComponentName(fiber.type) || 'Component',
+        );
+        didWarnAboutUnsafeLifecycles.add(fiber.type);
+      });
+
+      pendingComponentWillUpdateWarnings = [];
+    }
+
+    const UNSAFE_componentWillUpdateUniqueNames = new Set();
+    if (pendingUNSAFE_ComponentWillUpdateWarnings.length > 0) {
+      pendingUNSAFE_ComponentWillUpdateWarnings.forEach(fiber => {
+        UNSAFE_componentWillUpdateUniqueNames.add(
+          getComponentName(fiber.type) || 'Component',
+        );
+        didWarnAboutUnsafeLifecycles.add(fiber.type);
+      });
+
+      pendingUNSAFE_ComponentWillUpdateWarnings = [];
+    }
+
+    // Finally, we flush all the warnings
+    // UNSAFE_ ones before the deprecated ones, since they'll be 'louder'
+    if (UNSAFE_componentWillMountUniqueNames.size > 0) {
+      const sortedNames = setToSortedString(
+        UNSAFE_componentWillMountUniqueNames,
+      );
       warningWithoutStack(
         false,
-        'Expected to find a StrictMode component in a strict mode tree. ' +
-          'This error is likely caused by a bug in React. Please file an issue.',
+        'Using UNSAFE_componentWillMount in strict mode is not recommended and may indicate bugs in your code. ' +
+          'See https://fb.me/react-async-component-lifecycle-hooks for details.\n\n' +
+          '* Move component logic from UNSAFE_componentWillMount into componentDidMount (preferred in most cases) ' +
```

Comment:
I don't follow this one. 
"Move this code from UNSAFE_componentWillMount to componentDidMount (preferred in most cases) or the constructor." 
Do you think this would encourage people to put side effectful code into the constructor? I don't think so because they'll try componentDidMount before that.

Reply:
My concern was that this sentence:
  "_Move this code from UNSAFE_componentWillMount to componentDidMount (preferred in most cases) or the constructor._"

Might sometimes be read and interpreted as this:
  "_Move this code from UNSAFE_componentWillMount to componentDidMount or the constructor (preferred in most cases)._"

And the latter is not preferable.

Reply:
I like the new wording btw.